src/ctap/clock.rs

```diff
@@ -0,0 +1,48 @@
+pub use embedded_time::Clock;
+use libtock_drivers::result::FlexUnwrap;
+use libtock_drivers::timer::Timer;
+
+pub struct LibtockClock<'a, const CLOCK_FREQUENCY: u32>(Timer<'a>);
+impl<'a, const CLOCK_FREQUENCY: u32> LibtockClock<'a, CLOCK_FREQUENCY> {
+    pub fn new(timer: Timer<'a>) -> Self {
+        Self(timer)
+    }
+}
+
+#[cfg(target_pointer_width = "32")]
+pub type ClockInt = u32;
+#[cfg(target_pointer_width = "64")]
```

Comment:
As discussed with @kaczmarczyck in the beginning of the project, we expect the future crate can also be used for linux kernel as an example. In the current set up, it is used for desktop tests. I do not understand the `feature="std"` part. Could you explain what is the relation with `ClockInt`?

Reply:
I think all your recent comments are related to the same issue. So I'll answer here with comments within how the `clock` module could look like (I'm omitting some parameters for readability):

```rust
#[cfg(not(feature = "std"))] // enabled for production (currently only Tock is supported)
mod prod {
    use libtock_drivers;
    pub struct LibtockClock(Timer);
    impl Libtock { pub fn new() { ... } }
    impl Clock for Libtock {
        type T = u32;  // ideally this should be a custom 24-bits integer
        ...
    }
}

#[cfg(feature = "std")] // enabled for testing and fuzzing
mod test {
    // We don't need anything from libtock for testing. We shouldn't even compile it.
    pub struct TestClock;
    impl Clock for TestClock { ... }
}

#[cfg(not(feature = "std"))]
pub type CtapClock = prod::LibtockClock;
#[cfg(feature = "std")]
pub type CtapClock = test::TestClock;
```

As you can see, `feature = "std"` means testing (including fuzzing which is a form of testing) and `not(feature = "std")` means production. We currently only support Tock for production so there's no need to have this `ClockInt = u64`. This would currently be dead-code and in the future irrelevant because hidden behind the hardware abstraction. In particular the CTAP library will only know of `H::Clock::T` and the user would be responsible for defining this type. In particular, in the code above, you can see the code in the `prod` and `test` module as if it were written by the user of the CTAP library. In one case (`prod`) they do `T = u32` or `T = u24` or if within linux they could do `T = u64`. In the other case (`test`) they do `T = u64`.

Does this make more sense?